### PR TITLE
Add a HasLink instance for Auth

### LINKED ISSUE #105: Should servant-auth provide a HasLink instance for the Auth combinator ?
Thanks for providing this great library!  However I'm having a little trouble `safeLink` on protected routes.

Having an `Auth` combinator means that route is not a `HasLink` instance, because the `Auth` combinator does not provide its own `HasLink` instance.

It _might_ be possible to write an orphan instance when needed and use it, ( _might_ as in ' I've found this surprisingly difficult to do'), but having servant-auth provide this would remove the necessity for doing so.

So
-  should servant-auth provide this ?
-  if not, can someone provide a snippet showing a correctly functioning orphan instance ?

I've tried the following  (servant-0.13, servant-auth-0.3.2)

```haskell
{-# LANGUAGE DataKinds           #-}
{-# LANGUAGE FlexibleContexts    #-}
{-# LANGUAGE FlexibleInstances   #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE TypeFamilies        #-}
{-# LANGUAGE TypeOperators       #-}
{-# OPTIONS_GHC -fno-warn-orphans #-}
module Servant.Example.Links
   ( Auth
   )
where

import           Data.Proxy   (Proxy (..))
import           Servant
import           Servant.Auth (Auth)

-- Similar to the instance for AuthProtect in Servant.Links
instance HasLink sub => HasLink (Auth (tag :: [*]) :> sub) where
  type MkLink (Auth (tag :: [*]) :> sub) = MkLink sub
  toLink _ = toLink (Proxy :: Proxy sub)
```

This compiles but trying to use it does not - it is having the effect I was expecting. calling `safeLink` with a simple `Auth` protected route still fails as follows:
```
λ> safeLink (Proxy :: Proxy MyAPI) (Proxy :: Proxy ProtectedRoute)

<interactive>:18:1: error:
    • No instance for (HasLink ProtectedRoute)
        arising from a use of ‘safeLink’
    • In the expression:
        safeLink (Proxy :: Proxy MyAPI) (Proxy :: Proxy ProtectedRoute)
      In an equation for ‘it’:
          it = safeLink (Proxy :: Proxy MyAPI) (Proxy :: Proxy ProtectedRoute)
```

### PR DESCRIPTION
**Pull request by adetokunbo**
Fixes #105

### CODE REVIEW

**Comment by domenkozar** [comment]
```
src/Servant/Auth.hs:20:8: error:
    • Number of parameters must match family declaration; expected 2
    • In the type instance declaration for ‘MkLink’
      In the instance declaration for
        ‘HasLink (Auth (tag :: [*]) value :> sub)’
   |
20 |   type MkLink (Auth (tag :: [*]) value :> sub) = MkLink sub
   |        ^^^^^^
```

**Comment by adetokunbo** [comment]
@domenkozar @phadej : sorry for the delay.

This initially failed on CI because I had built it against an older version of servant with an earlier definition of MkLinks.

It is difficult to tell the precise version where MkLinks changed from servant's Changelog - what is the correct lower-bound for the servant dependency ?

**Comment by adetokunbo** [comment]
@domenkozar : I just noticed an email from you about the earlier CI breakage, so just to confirm, the CI build is now OK

File: servant-auth/src/Servant/Auth.hs [hunk 1]
```diff
@@ -1,12 +1,25 @@
+{-# LANGUAGE CPP                        #-}
```
Comment (domenkozar) [inline servant-auth/src/Servant/Auth.hs]: Don't think it needs CPP?

Comment (adetokunbo) [reply servant-auth/src/Servant/Auth.hs]: Sorry, I've removed that

**Comment by domenkozar** [comment]
I think this works only with servant 0.14, but we can revise that later, it's not clear to me how much compatibility there should be kept between servant libraries, although servant-auth is a prime citizen.

**Pull request merged**